packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -184,8 +158,6 @@ export const useGridSelection = (apiRef: GridApiRef): void => {
   );
 
   useGridApiEventHandler(apiRef, GRID_ROW_CLICK, handleRowClick);
-  useGridApiOptionHandler(apiRef, GRID_ROW_SELECTED, onRowSelected);
```

Comment:
Don't forget to remove this event from the constants file and to list it as a breaking change here.